plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx

```diff
@@ -14,47 +15,77 @@ import { WC_ASSET_URL } from '~/utils/admin-settings';
 import './incentive-banner.scss';
 import { StatusBadge } from '~/settings-payments/components/status-badge';
 
-export const IncentiveBanner = () => {
+interface IncentiveBannerProps {
+	/**
+	 * Incentive data.
+	 */
+	incentive: PaymentIncentive;
+	/**
+	 * Callback used when an incentive is accepted.
+	 *
+	 * @param id   Plugin ID.
+	 * @param slug Plugin slug.
+	 */
+	onAccept: ( id: string, slug: string ) => void;
+	/**
+	 * Callback to handle dismiss action.
+	 *
+	 * @param dismissUrl Dismiss URL.
+	 * @param context    The context in which the incentive is dismissed. (e.g. whether it was in a modal or banner).
+	 */
+	onDismiss: ( dismissUrl: string, context: string ) => void;
+}
+
+export const IncentiveBanner = ( {
+	incentive,
+	onDismiss,
+	onAccept,
+}: IncentiveBannerProps ) => {
 	const [ isSubmitted, setIsSubmitted ] = useState( false );
-	const [ isDismissClicked, setIsDismissClicked ] = useState( false );
+	const [ isDismissed, setIsDismissed ] = useState( false );
+	const [ isBusy, setIsBusy ] = useState( false );
+
+	const incentiveContext = 'wc_settings_payments__banner';
 
-	const handleSetup = () => {
+	const handleAccept = () => {
+		setIsBusy( true );
+		onAccept( 'woopayments', 'woocommerce-payments' );
```

Comment:
It's the same TODO for both. The problem we have is that we currently don't detect _which_ incentive is being applied (as in, the gateway it belongs to), because they all belong to WooPayments right now. I can fix it, but will need to do some refactors so I'll address it separately.

Reply:
okay, in the next PR!